### PR TITLE
Added example for middleware language detection

### PR DESCRIPTION
**Pull request by obrassard**
Created an example for `examples/with-preferred-language-middleware/`

I started from the **basic** example and added the `middleware.ts` and a `readme`.

### CODE REVIEW

File: examples/with-preferred-language-middleware/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const urlParts = url.split('/')
+  const dynamicRegex = /^\[\w+\]$/
+  for (let i = 0; i < schemaParts.length; i++) {
+    if (schemaParts[i].match(dynamicRegex)) {
+      const paramName = schemaParts[i].slice(1, -1)
+      dynamicSegments[paramName] = urlParts[i]
+    }
+  }
+  return dynamicSegments
+}
```
Comment (obrassard) [inline examples/with-preferred-language-middleware/middleware.ts:82]: @svobik7 Is there already a function in next-roots that we could use to do this ?

Comment (svobik7) [reply examples/with-preferred-language-middleware/middleware.ts:82]: Unfortunately, I dont think so. Let's keep it here.

**Comment by svobik7** [comment]
Works great. Thank you for this PR. Will add you to the CONTRIBUTORS.md (once I add this file :-D)

**Comment by obrassard** [comment]
You're welcome !

**Pull request merged**